extensions/standard-processors/processors/ReplaceText.cpp

```diff
@@ -0,0 +1,359 @@
+core::annotation::Input ReplaceText::getInputRequirement() const {
+  return core::annotation::Input::INPUT_REQUIRED;
+}
+
+void ReplaceText::initialize() {
+  setSupportedProperties({
+      SearchValue,
+      ReplacementValue,
+      MaximumBufferSize,
+      ReplacementStrategy,
+      EvaluationMode,
+      LineByLineEvaluationMode
+  });
+  setSupportedRelationships({
+      Success,
+      Failure
+  });
+}
+
+void ReplaceText::onSchedule(const std::shared_ptr<core::ProcessContext>& context, const std::shared_ptr<core::ProcessSessionFactory>&) {
+  gsl_Expects(context);
+
+  const std::optional<std::string> evaluation_mode = context->getProperty(EvaluationMode);
+  evaluation_mode_ = EvaluationModeType::parse(evaluation_mode.value().c_str());
+  logger_->log_debug("the %s property is set to %s", EvaluationMode.getName(), evaluation_mode_.toString());
+
+  const std::optional<std::string> line_by_line_evaluation_mode = context->getProperty(LineByLineEvaluationMode);
+  if (line_by_line_evaluation_mode) {
+    line_by_line_evaluation_mode_ = LineByLineEvaluationModeType::parse(line_by_line_evaluation_mode->c_str());
+    logger_->log_debug("the %s property is set to %s", LineByLineEvaluationMode.getName(), line_by_line_evaluation_mode_.toString());
+  }
+
+  const std::optional<std::string> replacement_strategy = context->getProperty(ReplacementStrategy);
+  replacement_strategy_ = ReplacementStrategyType::parse(replacement_strategy.value().c_str());
+  logger_->log_debug("the %s property is set to %s", ReplacementStrategy.getName(), replacement_strategy_.toString());
+
+  context->getProperty(MaximumBufferSize.getName(), maximum_buffer_size_);
+  logger_->log_debug("the %s property is set to %" PRIu64 " bytes", MaximumBufferSize.getName(), maximum_buffer_size_);
+}
+
+void ReplaceText::onTrigger(const std::shared_ptr<core::ProcessContext>& context, const std::shared_ptr<core::ProcessSession>& session) {
+  gsl_Expects(context);
+  gsl_Expects(session);
+
+  std::shared_ptr<core::FlowFile> flow_file = session->get();
+  if (!flow_file) {
+    logger_->log_trace("No flow file");
+    return;
+  }
+
+  readSearchValueProperty(context, flow_file);
+  readReplacementValueProperty(context, flow_file);
+
+  switch (evaluation_mode_.value()) {
+    case EvaluationModeType::ENTIRE_TEXT:
+      replaceTextInEntireFile(flow_file, session);
+      return;
+    case EvaluationModeType::LINE_BY_LINE:
+      replaceTextLineByLine(flow_file, session);
+      return;
+  }
+
+  throw Exception{PROCESSOR_EXCEPTION, utils::StringUtils::join_pack("Unsupported ", EvaluationMode.getName(), ": ", evaluation_mode_.toString())};
+}
+
+void ReplaceText::readSearchValueProperty(const std::shared_ptr<core::ProcessContext>& context, const std::shared_ptr<core::FlowFile>& flow_file) {
+  bool found_search_value;
+  if (replacement_strategy_ == ReplacementStrategyType::REGEX_REPLACE) {
+    found_search_value = context->getProperty(SearchValue.getName(), search_value_);
```

Comment:
won't writing members (`search_value_`, `search_regex_`) interferes with other threads of this same processor?

Reply:
Good point, `onTrigger()` was not thread-safe.  Fixed in https://github.com/apache/nifi-minifi-cpp/pull/1170/commits/86eb3f776005301a6507ca7b27e5d28736d5923a.